src/test/java/com/microsoft/sqlserver/jdbc/datatypes/SQLVariantTest.java

```diff
@@ -0,0 +1,734 @@
+/*
+ * Microsoft JDBC Driver for SQL Server
+ * 
+ * Copyright(c) Microsoft Corporation All rights reserved.
+ * 
+ * This program is made available under the terms of the MIT License. See the LICENSE file in the project root for more information.
+ */
+package com.microsoft.sqlserver.jdbc.datatypes;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.sql.CallableStatement;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.Arrays;
+
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.platform.runner.JUnitPlatform;
+import org.junit.runner.RunWith;
+
+import com.microsoft.sqlserver.jdbc.SQLServerBulkCopy;
+import com.microsoft.sqlserver.jdbc.SQLServerConnection;
+import com.microsoft.sqlserver.jdbc.SQLServerException;
+import com.microsoft.sqlserver.jdbc.SQLServerPreparedStatement;
+import com.microsoft.sqlserver.jdbc.SQLServerResultSet;
+import com.microsoft.sqlserver.testframework.AbstractTest;
+import com.nimbusds.oauth2.sdk.ParseException;
+
+/**
+ * Tests for supporting sqlVariant
+ *
+ */
+@RunWith(JUnitPlatform.class)
+public class SQLVariantTest extends AbstractTest {
+
+    static SQLServerConnection con = null;
+    static Statement stmt = null;
+    static String tableName = "SqlVariant_Test";
+    static String inputProc = "sqlVariant_Proc";
+    static String procedureName = "TVP_SQLVariant_Proc";
+
+    @Test
+    public void readInt() throws SQLException, SecurityException, IOException {
+        int value = 2;
+        createAndPopulateTable("int", value);
+        SQLServerResultSet rs = (SQLServerResultSet) stmt.executeQuery("SELECT * FROM " + tableName);
+        while (rs.next()) {
```

Comment:
for all the test cases, instead of using `while (rs.next())`, I think we should use `rs.next()`. With while() block, if the table is empty somehow, the assertEquals wont be checked.